### DISCUSSION TITLE
Deleting Indexes that has value equal to a particular metadata value.

### DISCUSSION THREAD

**Discussion by Pranav-NG** [milvus-io/milvus#27001] [Q&A and General discussion] [answered]
Hi Team , 

I am building a data pipeline that updates the data in Milvus often.
It should happen in the way in the way that , the old data (whose metadata==value) gets deleted and new data gets ingested.

Ideally , I want it to look like - 
1. List all values in a metadata column.
2. List the indexes whose metadata == particular value (This value is a string)
3. Delete the indexes.

Please guide on this .

Thanks.

**Comment by yhmo** [comment, marked as answer]
"List all values in a metadata column" ----------- to fetch values from a collection, use query() interface.
Assume the primary key field name is "id":
```
results = collection.query(expr="id >= 0", outpur_fields=["meta_field_name"])
metas = []
for res in results:
    metas.append(res["meta_field_name"])
```
But the query() interface can only return at most 16384 results, you can add a configuration in the milvus.yaml to enlarge the limitation:
```
quotaAndLimits:
  limits:
    maxQueryResultWindow: 1000000
```

"List the indexes whose metadata == particular value" ---------------- also use query() interface
Assume the metadata column name is "title":
```
results = collection.query(expr='title =="particular_value"', outpur_fields=["*"])
for res in results:
    print(res)
```

"Delete the indexes" --------- use delete() interface https://milvus.io/docs/delete_data.md
